pkg/reconcilers/deployment.go

```diff
@@ -0,0 +1,152 @@
+package reconcilers
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/3scale/apicast-operator/pkg/k8sutils"
+
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+// DCMutateFn is a function which mutates the existing Deployment into it's desired state.
+type DCMutateFn func(desired, existing *appsv1.Deployment) bool
```

Comment:
Could we change the `DCMutateFn` name of the type to DeploymentMutateFn?. In the case of APIcast operator we are not dealing with DeploymentConfigs (DC) but with Deployments

Reply:
done